guide/event-handling/README.md

```diff
@@ -0,0 +1,284 @@
+
+In order to listen for events, we have to register them first. In other words, we have to tell nodejs what is the name of the event that we want to listen for and what function should be called, when that event emits. Since this function gets called when an event emits, we say it's a [callback function](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function). This is done by using the `on` or `once` method of an `EventEmitter`. The `on` method is used to register callback function for events that can emit multiple times, while functions registered using `once` method get unregistered after a single event.
+
+::: tip
+You can learn more about `EventEmitter` [here](https://nodejs.org/api/events.html#events_class_eventemitter)
+:::
+
+The `Client` class in discord.js extends this `EventEmitter` class. Therefore, the `client` object also has these `on` and `once` methods that you can use to register events. These methods take two arguments: name of the event and a callback function. You get the name of the event from `event.name`.
+
+The callback function you passed, takes argument(s) returned by its respective event, collects them in an `args` array using the `...` rest parameter, then passes the elements in the `args` array to the `event.execute` function using the `...` spread syntax and then finally calls it. Even though `rest` and `spread` look the same, their functionality is exactly opposite of each other. You have to use them here because different types of events in discord.js return different number of arguments. The `rest` parameter collects these unknown number of arguments into a single array, the `spread` operator then takes each element from this `args` array and passes it to the `execute` function.
+
+::: tip
+Learn more about `rest` and `spread` [here](https://javascript.info/rest-parameters-spread)
+:::
+
+Now, lets come back to the event-handler and see what's happening in its last few lines:
+
+```js
+if (event.once) {
+	client.once(event.name, (...args) => event.execute(...args));
+} else {
+	client.on(event.name, (...args) => event.execute(...args));
+}
+```
+
+You can listen for an event either only once or multiple times by registering it using `once` or `on` respectively. In both of the above conditions, you are registering a callback function that will be called when its respective event emits, but for events that have their `once` property set to `true`, you will register them using the `once` method, so that it gets unregistered after a single event.
+
+Before you go any further you should check the event handler is working or not. Open your terminal, type `node index.js` and press enter. If you see `Ready!` gets printed, then congratulations, you've made an event handler. After this, listening for other events is as easy as creating a new event file in the `events` folder. The event handler will automatically retrieve this new event file and register the respective event whenever you restart the bot.
```

Comment:
```suggestion
In order to listen for events, you have to register them first. This is done by using the `on` or `once` methods of an `EventEmitter` instance. The `on` method for events that can emit multiple times, while `once` will run once and unregister the listener after a single emit.

::: tip
You can learn more about `EventEmitter` [here](https://nodejs.org/api/events.html#events_class_eventemitter).
:::

The `Client` class in discord.js extends the `EventEmitter` class. Therefore, the `client` object also has these `on` and `once` methods that you can use to register events. These methods take two arguments: name of the event and a callback function.

The callback function passed takes argument(s) returned by its respective event, collects them in an `args` array using the `...` [rest parameter syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters), then calls `event.execute` function while passing in the `args` array using the `...` [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). They are used here because different events in discord.js have different number of arguments. The rest parameter collects these variable number of arguments into a single array, and the spread syntax then takes these elements and passes it to the `execute` function.

After this, listening for other events is as easy as creating a new file in the `events` folder. The event handler will automatically retrieve and register it whenever you restart your bot.
```